src/core/unif.ml

```diff
@@ -32,13 +56,12 @@ let add_constr : problem -> constr -> unit = fun p c ->
    as well as the constraint [(c,a,b)] where [a] is the type of [t] and [b]
    the type of [u] if they can be infered. *)
 let add_unif_rule_constr : problem -> constr -> unit = fun p (c,t,u) ->
-  add_constr p (c,t,u);
   match Infer.infer_noexn p c t with
-  | None -> ignore (Infer.infer_noexn p c u)
-  | Some a ->
+  | None -> add_constr p (c, t, u)
```

Comment:
Shouldn't you keep the test that u is typable?

Reply:
What does it mean when `t` is not typable? What should we do in that case?

Reply:
Right. We should probably fail in this case.